testing/rspec/spec/unit/linters/scenario_with_bad_name_linter_unit_spec.rb

```diff
@@ -0,0 +1,108 @@
+require_relative '../../../../../environments/rspec_env'
+
+
+RSpec.describe CukeLinter::ScenarioWithBadNameLinter do
+
+  let(:model_file_path) { 'some_file_path' }
+
+  it_should_behave_like 'a linter at the unit level'
+
+  it 'has a name' do
+    expect(subject.name).to eq('ScenarioWithBadNameLinter')
+  end
+
+  describe 'linting' do
+
+    ['scenario', 'outline'].each do |model_type|
+
+      context "with a #{model_type} that has a bad name" do
+
+        context 'because its name contains test' do
+
+          let(:test_model) do
+            model      = CukeLinter::ModelFactory.send("generate_#{model_type}_model", parent_file_path: model_file_path)
+            model.name = 'Test models with bad names'
+
+            model
+          end
+
+          it_should_behave_like 'a linter linting a bad model'
+
+
+          it 'records a problem' do
+            result = subject.lint(test_model)
+
+            expect(result[:problem]).to eq('Prefer name your scenarios using "Given" and "When" rather than "test", "verify" or "check".')
+          end
+
+        end
+
```

Comment:
Could you dynamically generate these three tests similar to how they are all already in a loop that dynamically uses scenarios or outlines as needed? The only thing that I'm seeing that makes them different is the change of letter casing and, rather than ninja-ing in implicit requirements, I think that it would be better to have a separate spec that plainly declares that case sensitivity isn't used.